Packs/CommonScripts/Scripts/ParseEmailFilesV2/ParseEmailFilesV2_test.py

```diff
@@ -417,3 +419,83 @@ def test_md_output_with_body_text():
 
     md = data_to_md(email_data)
     assert expected == md
+
+
+@pytest.mark.parametrize('nesting_level_to_return, output, res', [('All files', ['output1'], ['output1']),
+                                                                  ('Outer file', ['output1', 'output2', 'output3'],
+                                                                   ['output1']),
+                                                                  ('Inner file', ['output1', 'output2', 'output3'],
+                                                                   ['output3'])])
+def test_parse_nesting_level(nesting_level_to_return, output, res):
+    """
+    Given:
+    - parsed email output, nesting_level_to_return param - All files.
+    - parsed email output, nesting_level_to_return param - Outer file.
+    - parsed email output, nesting_level_to_return param - Inner file.
+    When:
+    - Getting all nested emails.
+    - Getting only outer email file.
+    - Getting only inner email file.
+    Then: Validate that returned result as expected.
```

Comment:
This is twisted the `When should be `calling the parse_nesting_level function and the `Then` should be the 3 different scenarios you are asserting.